packages/picasso/src/LexicalEmojiPlugin/nodes/CustomEmojiNode.tsx

```diff
@@ -0,0 +1,153 @@
+  DOMExportOutput,
+  EditorConfig,
+  LexicalNode,
+  NodeKey,
+  SerializedElementNode,
+  Spread,
+} from 'lexical'
+
+export interface CustomEmojiPayload {
+  src: string
+  id: string
+}
+
+type SerializedCustomEmojiNode = Spread<
+  {
+    src: string
+    id: string
+  },
+  SerializedElementNode
+>
+
+const convertImageElement = (domNode: Node): null | DOMConversionOutput => {
+  if (domNode instanceof HTMLImageElement) {
+    const src = domNode.getAttribute('src')
+    const id = domNode.getAttribute('data-emoji-name')
+
+    if (src && id) {
+      return {
+        node: $createCustomEmojiNode({
+          src,
+          id,
+        }),
+      }
+    }
+
+    return null
+  }
+
+  return null
+}
+
+export class CustomEmojiNode extends DecoratorNode<JSX.Element> {
+  __src: string
+  __id: string
+
+  static getType() {
+    return 'custom-emoji'
+  }
+
+  static clone(node: CustomEmojiNode): CustomEmojiNode {
+    return new CustomEmojiNode(node.__src, node.__id)
+  }
+
+  constructor(src: string, id: string, key?: NodeKey) {
+    super(key)
+    this.__src = src
+    this.__id = id
+  }
+
+  static importJSON(
+    serializedNode: SerializedCustomEmojiNode
+  ): CustomEmojiNode {
+    const { src, id } = serializedNode
+
+    const node = $createCustomEmojiNode({ src, id })
+
+    return node
+  }
+
+  createDOM(config: EditorConfig): HTMLElement {
+    const span = document.createElement('span')
+
+    const theme = config.theme
+    const className = theme.customEmoji
+
+    if (className !== undefined) {
+      span.className = className
+    }
+
+    return span
+  }
+
+  // update() {}
+
+  updateDOM() {
+    return false
+  }
+
+  exportDOM(): DOMExportOutput {
+    const element = document.createElement('img')
+
+    element.setAttribute('src', this.__src)
+    element.setAttribute('data-src', this.__src)
+    element.setAttribute('data-emoji-name', this.__id)
+
+    return { element }
+  }
+
+  static importDOM(): DOMConversionMap | null {
+    return {
+      img: () => ({
+        conversion: convertImageElement,
+        priority: 0,
+      }),
+    }
+  }
+
+  exportJSON(): SerializedCustomEmojiNode {
+    return {
+      version: 1,
+      type: 'custom-emoji',
+      src: this.__src,
+      id: this.__id,
+      children: [],
+      direction: this.__direction,
+      format: '',
+      indent: this.__indent,
+    }
+  }
+
+  // isInline() {
+  //   return true
+  // }
+
+  // canIndent() {
+  //   return false
+  // }
+
+  decorate() {
+    return (
+      <img src={this.__src} data-src={this.__src} data-emoji-name={this.__id} />
+    )
+  }
+}
+
+export const $isCustomEmojiNode = (
+  node: LexicalNode | null | undefined
+): node is CustomEmojiNode => {
+  return node instanceof CustomEmojiNode
+}
+
+export const $createCustomEmojiNode = ({
+  src,
+  id,
+}: CustomEmojiPayload): CustomEmojiNode => {
+  // return new CustomEmojiNode(src, id)
+  return $applyNodeReplacement(new CustomEmojiNode(src, id))
+}
```

Comment:
I would strech Picasso code conventions to these methods as well. In fact it was a bit confusing at first glance, because of the `$` sign I thought these functions were from Lexical, when they are implemented on Picasso itself. I don't see why we would need to prefix them with dollar signs, as it only makes sense when internally used by Lexical, wdyt?
```suggestion
export const isCustomEmojiNode = (
  node: LexicalNode | null | undefined
): node is CustomEmojiNode => {
  return node instanceof CustomEmojiNode
}

export const createCustomEmojiNode = ({
  src,
  id,
}: CustomEmojiPayload): CustomEmojiNode => {
  // return new CustomEmojiNode(src, id)
  return $applyNodeReplacement(new CustomEmojiNode(src, id))
}
```

Reply:
@augustobmoura it is best practice to prefix functions with `$` when they can be used only in `editor.update` or `editor.read`

Reply:
Well, this is a Lexical coding convention, I'm personally not a fan of [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation), and external contributors could find using the dollar sign confusing as well. If we decide on using it only for Lexical updates we need to document this somewhere, and probably have a discussion with the team, otherwise functions definitions will be inconsistent.

Let's leave this how it is and discuss it with the team tomorrow on the daily, is not a priority